controllers/storagecluster/backingstorageclasses.go

```diff
@@ -29,41 +30,49 @@ func (obj *backingStorageClasses) ensureCreated(r *StorageClusterReconciler, sc
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	existingBackingStorageClasses, err := getBackingStorageClasses(r)
+	existingBackingStorageClasses := &v1.StorageClassList{}
+	err = r.Client.List(r.ctx,
+		existingBackingStorageClasses,
+		client.MatchingLabels(map[string]string{
+			backingStorageClassLabel: strconv.FormatBool(true),
+		}),
+	)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	existingStorageClasses := make(map[string]*v1.StorageClass)
```

Comment:
No need to call make?
Also the name of the var should reflect the usage.
```suggestion 
	existingStorageClassesByName := map[string]*v1.StorageClass{}
```